### PR TITLE
fix(threat) tolerate empty namespace URI in namespace declaration

### PR DESCRIPTION
**Pull request by hanshuebner**

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Copyright (C) 2003-2007 The Kepler Project, 2013-2022 Matthew Wild
```
Comment (alerque) [inline LICENSE:1]: Date-bumping the license is fine, but lets do it in a different commit and maybe even PR, it is a housekeeping thing and the git blame situation gets really noisy if important commits that might have unintended consequences also touch a bunch of unrelated bits.

File: rockspecs/luaexpat-1.5.2-1.rockspec [hunk 2]
```diff
@@ -0,0 +1,51 @@
+local package_name = "luaexpat"
```
Comment (alerque) [inline rockspecs/luaexpat-1.5.2-1.rockspec:1]: Please don't include new rockspecs or version bumps in this PR. A release may be warranted but the release process is actually going to be harder to manage with this in an unrelated PR than it would have been otherwise. If there are changes that need to be made to the rockspec to go along with the PR make it on the dev/scm rockspec instead. We'll propagate that on the next release.

File: src/lxplib.h [hunk 3]
```diff
@@ -2,8 +2,8 @@
 ** See Copyright Notice in license.html
 */
 
-#define LuaExpatCopyright	"Copyright (C) 2003-2007 The Kepler Project, 2013-2022 Matthew Wild"
```
Comment (alerque) [inline src/lxplib.h:5]: Again lets not mix housekeeping, release activities, and feature/fixes in the same commit & PR.

**Review by alerque — CHANGES_REQUESTED** [review]

**Comment by hanshuebner** [comment]
@alerque Makes total sense, sorry for not thinking about it myself.

**Comment by hanshuebner** [comment]
@alerque @Tieske I'd appreciate a review & merge.  I've got the release stuff on a separate branch that I'll open a PR for once this one is merged.

File: src/lxp/threat.lua [hunk 4]
```diff
@@ -376,7 +376,7 @@ function threat.new(callbacks, separator, merge_character_data)
 					return threat_error("prefix too long")
 				end
 
-				if checks.namespaceUri and #namespaceUri > checks.namespaceUri then
+				if checks.namespaceUri and namespaceUri and #namespaceUri > checks.namespaceUri then
```
Comment (alerque) [inline src/lxp/threat.lua:379]: Can you elucidate a bit on why this change is valid/needed? I am far from an XML expert but it seems counter intuitive to me that we're both specifically being asked to check the namespaceUri (if checks.namespaceUri) and also allowing nil as a valid number that is greater than the number we're supposed to check for. I'm not saying I know this approach is invalid, but I think it should be annotated somehow so it is clear *why* it is this way.

Comment (hanshuebner) [reply src/lxp/threat.lua:379]: XML has a default namespace, which can be set using `<foo xmlns="namespace-uri">` and namespace prefixes which can be set using `<foo xmlns:bar="namespace-uri">`.  Namespace URIs can be any valid URI or empty.  Thus, by using `<foo xmlns="">`, it is possible to reset the default namespace to none. See https://www.w3.org/TR/xml-names/#defaulting

**Comment by Tieske** [comment]
I added #39 with an additional test
